internal/workflow_testsuite.go

```diff
@@ -64,17 +71,40 @@ type (
 	}
 )
 
+func newEncodedValues(values []byte, dc encoded.DataConverter) encoded.Values {
+	if dc == nil {
+		dc = newDefaultDataConverter()
+	}
+	return &EncodedValues{values, dc}
+}
+
 // Get extract data from encoded data to desired value type. valuePtr is pointer to the actual value type.
 func (b EncodedValues) Get(valuePtr ...interface{}) error {
-	if b == nil {
+	if !b.HasValues() {
 		return ErrNoData
 	}
-	return getHostEnvironment().decode(b, valuePtr)
+	return b.dataConverter.FromData(b.values, valuePtr...)
 }
 
 // HasValues return whether there are values encoded.
 func (b EncodedValues) HasValues() bool {
-	return b != nil
+	return b.values != nil
+}
+
+// Get extract data from encoded data to desired value type. valuePtr is pointer to the actual value type.
+func (b ErrorDetailsValues) Get(valuePtr ...interface{}) error {
+	if !b.HasValues() {
+		return ErrNoData
+	}
+	for i, item := range b {
```

Comment:
maybe do a sanity check to make sure len(valuePtr) <= len(b) and then loop through valuePtr instead of b.

Reply:
changed